packages/lib/server/repository/user.ts

```diff
@@ -252,6 +252,20 @@ export class UserRepository {
     return user;
   }
 
+  static async findByIdWithOptionalSelect({ id, select }: { id: number; select?: Prisma.UserSelect }) {
+    const user = await prisma.user.findUnique({
+      where: {
+        id,
+      },
+      select: { ...select },
```

Comment:
`{ ...select }` is need to prevent type errors

Reply:
@keithwillcode or @hariombalhara thoughts on this? I remember bringing this up before, and we decided to just return the whole user object. 

Is it because the `select` param adds a dependency to Prisma, which defeats the purpose of having the repository pattern? Law of Demeter?